store/cachekv/store.go

```diff
@@ -68,38 +92,38 @@ func (store *Store) Get(key []byte) (value []byte) {
 }
 
 // Set implements types.KVStore.
-func (store *Store) Set(key, value []byte) {
+func (store *GStore[V]) Set(key []byte, value V) {
 	types.AssertValidKey(key)
-	types.AssertValidValue(value)
+	types.AssertValidValueGeneric(value, store.isZero, store.valueLen)
 
 	store.mtx.Lock()
 	defer store.mtx.Unlock()
 	store.setCacheValue(key, value, true)
 }
 
 // Has implements types.KVStore.
-func (store *Store) Has(key []byte) bool {
+func (store *GStore[V]) Has(key []byte) bool {
 	value := store.Get(key)
-	return value != nil
+	return !store.isZero(value)
 }
 
 // Delete implements types.KVStore.
-func (store *Store) Delete(key []byte) {
+func (store *GStore[V]) Delete(key []byte) {
 	types.AssertValidKey(key)
 
 	store.mtx.Lock()
 	defer store.mtx.Unlock()
 
-	store.setCacheValue(key, nil, true)
+	store.setCacheValue(key, store.zeroValue, true)
 }
 
-func (store *Store) resetCaches() {
+func (store *GStore[V]) resetCaches() {
 	if len(store.cache) > 100_000 {
```

Comment:
potentially worthwhile  to investigate this - who knows if there is empirical data for this or things have changed overtime.  Outside of this scope of course